ivy/functional/ivy/general.py

```diff
@@ -2177,15 +2177,12 @@ def stable_pow(
 
 
 @handle_exceptions
-def get_all_arrays_in_memory():
+def get_all_arrays_in_memory() -> ivy.Array:
     """Gets all arrays which are currently alive."""
     all_arrays = list()
     for obj in gc.get_objects():
```

Comment:
Thank you I'll look into it

Reply:
Yes when it the backend is `numpy` and `""`

Reply:
Will this work 

    for obj in gc.get_objects():
        try:
            if isinstance(o,ivy.Array) and ivy.is_native_array(obj) :
                all_arrays.append(obj)
        except Exception:
            pass

Reply:
No this will throw the same error as first one
What I would recommend is adding another try block for that condition I mentioned and keep the existing one as else part

Reply:
```python3

if ivy.current_backend_str() == "numpy" or ivy.current_backend_str() == "":
        for obj in gc.get_objects():
            try:
                if ivy.is_ivy_array(obj):
                    all_arrays.append(obj)
            except Exception:
                pass
        return all_arrays
    for obj in gc.get_objects():
        try:
            if ivy.is_native_array(obj):
                all_arrays.append(obj)
        except Exception:
            pass
```
I was thinking of something like this😊

Reply:
> ```python
> if ivy.current_backend_str() == "numpy" or ivy.current_backend_str() == "":
>         for obj in gc.get_objects():
>             try:
>                 if ivy.is_ivy_array(obj):
>                     all_arrays.append(obj)
>             except Exception:
>                 pass
>         return all_arrays
>     for obj in gc.get_objects():
>         try:
>             if ivy.is_native_array(obj):
>                 all_arrays.append(obj)
>         except Exception:
>             pass
> ```
> 
> I was thinking of something like thisblush

I mean can't we do it with single iteration?

Reply:
> This will return empty list when ivy arrays are stored in memory so for numpy backend and default backend ivy arrays are also stored. If you find a one iteration to handle this case as well it will be nice and may above example is a naive solution which covers both cases

```
def get_all_arrays_in_memory() -> ivy.Array:
    """Gets all arrays which are currently alive."""
    all_arrays = list()
    for obj in gc.get_objects():
        try:
            if ivy.current_backend_str() in ['','numpy']:
                if ivy.is_ivy_array(obj):
                    all_arrays.append(obj)
            else:
                if ivy.is_native_array(obj):
                    all_arrays.append(obj)

        except Exception:
            pass
    return all_arrays

```
You mean like this? do we really need two try_except block?

Reply:
Yea this one can work👍